### DISCUSSION TITLE
Splink with MS SQL Server

### DISCUSSION THREAD

**Discussion by JonathanBailiss** [moj-analytical-services/splink#1935] [Q&A] [answered]
Hello,

I work within a Local Authority and our toolset is quite constrained.  Our datasets are primarily held on an on-premise SQL Server.

Can Splink natively work with a SQL Server back-end and, if so, are there any code samples that would set me in the right direction?

If not would the recommended approach be to create a data flow from SQL Server into DuckDB and then use DuckDB for the linking?

At the moment I'm pretty much on the start line with this and haven't moved beyond confirming that the quickstart code sample executes correctly (it does!).

Apologies if this question has already been asked - I did look through the discussions first but may have overlooked it.

Thanks

**Comment by RobinL** [comment, marked as answer]
Hi Jonathan,

In a nutshell, yes, we'd recommend extracting the data from SQL Server and doing the linkage using DuckDB.

We have tried using MS SQL Server in the past for linkage (even getting as far as a working demo), but it suffers from two problems:
- The lack of fuzzy matching functions
- It's slow for the operations we need to use for Splink

DuckDB on the other hand is highly optimised for Splink-like workloads, and natively supports all the main fuzzy matching functions.

We're always happy to support colleagues in local authorities so do feel free to reach out.  My email is robinlinacre@hotmail.com (i'll reply from my work email, but don't want to post that here).
